File: Angular/src/app/models/conversation.model.ts
import { ApplicationUserModel } from './application-user.model';
import { MessageModel } from './message.model';

export class ConversationModel {
    conversationId: number;
    lastMessage: MessageModel;
    users: ApplicationUserModel[];
    wasRead: boolean;
    constructor(item: any) {
        this.conversationId = item.conversationId;
        this.lastMessage = item.lastmessage;
        this.users = item.users;
        this.wasRead = true;
        }
}

File: ASP.NET API/WAVC_WebApi/Models/HelperModels/ConversationModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WAVC_WebApi.Models
{
    public class ConversationModel
    {
        public int ConversationId { get; set; }
        public bool WasRead { get; set; }
        public List<ApplicationUserModel> Users { get; set; }
    }
}
